rfcs/45-stageless.md

```diff
@@ -2,488 +2,457 @@
 
-The most important way we can configure systems is by telling the scheduler *when* they should be run.
-The ordering of systems is always defined relative to other systems: either directly or by checking the systems that belong to a set.
-
-Applying an ordering constraint between sets causes ordering constraints to be created between all individual members of those sets.
-If an ordering is defined relative to an empty set, it will have no effect, and a warning will be emitted.
-This relatively gentle failure mode is important to ensure that plugins can order their systems with relatively strong assumptions that the default system sets exist, but continue to (mostly) work if those systems or sets are not present.
-
-In addition to the `.before` and `.after` methods, you can use **system chains** to create very simple linear dependencies between the successive members of an array of systems.
-(Note to readers: this is not the same as "system chaining" in Bevy 0.7 and earlier: that concept has been renamed to "system piping".)
-
-```rust
-fn main(){
-   App::new()
-   // These systems are connected using a string of ordering constraints
-
-   .add_systems((compute_attack, 
-                 compute_defense,
-                 check_for_crits,
-                 compute_damage,
-                 deal_damage,
-                 check_for_death)
-                 .chain()
-                 // We can configure all systems in the chain at once
-                 .in_set(GameSet::Combat)
-   )
-   .run()
-}
-```
-
-#### Run criteria
-
-While ordering constraints determine *when* a system will run, **run criteria** will determine *if* it will run at all.
-Each system can have any number of run criteria, which read data from the world to control whether or not that system runs.
-If (and only if) all of its run criteria return `true`, the system will run.
-If any of the run criteria are `false`, the system will be skipped.
-Systems that are skipped are considered completed for the purposes of ordering constraints.
-Run criteria are not systems, but can be created from systems which can read (but not write) data from the `World` and returns a boolean value.
-
-You can specify run criteria in several different ways:
+Conditions are each evaluated *at most once* during a single pass of a schedule. The evaluation occurs literally just before the system (or the first system encountered that is part of the set of interest) would be run. Because of this, results are guaranteed to be up-to-date. The state read by the conditions cannot be modified in an observable way before the guarded system starts.

```

Comment:
```suggestion
Conditions are each evaluated *at most once* during a single pass of a schedule.
The evaluation occurs just before the system (or the first system encountered that is part of the set of interest) would be run.
Because of this, results are guaranteed to be up-to-date. The state read by the conditions cannot be modified in an observable way before the guarded system starts.
```